src/multiselect_draggable.js

```diff
@@ -0,0 +1,336 @@
+   * that will have an event listener removed from
+   * @private
+   */
+  removePointerDownEventListener_(subDraggable) {
+    if (subDraggable) {
+      // Retrieve the stored handler function
+      const handler = subDraggable.pointerDownHandler;
+      if (handler) {
+        subDraggable.getSvgRoot().removeEventListener('pointerdown', handler);
+
+        // Clean up the stored handler reference
+        delete subDraggable.pointerDownHandler;
+      }
+    }
+  }
+
+  /**
+   * The handler for the pointer down event that mitigates
+   * the issue of setStart[draggable] overwriting the call that
+   * passes the multidraggable to Blockly.common.SetSelected().
+   * @param {PointerEvent} event A pointer down event
+   * @private
+   */
+  pointerDownEventHandler_(event) {
+    if (!inMultipleSelectionModeWeakMap.get(this.workspace)) {
+      Blockly.common.setSelected(this);
+    }
+  }
+
+  /**
+   * A method that determines whether the object is movable or not.
+   * @returns {boolean} Returns true as it is always movable
+   */
+  isMovable() {
+    if (inMultipleSelectionModeWeakMap.get(this.workspace)) {
+      return false;
+    }
+    return true;
+  }
+
+  /**
+   * Returns the current location of the multiselectDraggable
+   * in workspace coordinates.
+   * @returns {Blockly.utils.Coordinate} The coordinate of the current location
+   * on workspace (top left corner of multiselect-draggable)
+   */
+  getRelativeToSurfaceXY() {
+    return this.loc;
+  }
+
+  /**
+   * Starts a drag on the multiselectDraggable object if there
+   * are no other plugins that overwrite this method.
+   * It delegates the startDrag methods to the topmost subdraggables.
+   * @param {Blockly.Events.BLOCK_DRAG} e A drag event
+   */
+  startDrag(e) {
+    Blockly.Events.setGroup("asdfasdf")
+    for (const draggable of this.subDraggables) {
+      if (draggable[0] instanceof
+          Blockly.BlockSvg) {
+        // Save only the topmost subdraggables to do the drag on
+        // or if it is not any other draggable with parents/child
+        // relationships like workspace comments
+        if (!(hasSelectedParent(draggable[0], true))) {
+          this.topSubDraggables.push(draggable[0]);
+        }
+
+        // Save any connections between blocks of the same level
+        // (next statement)
+        if (!draggable[0].isShadow()) {
+          const parentBlock = draggable[0].getParent();
+          if (parentBlock && this.subDraggables.has(parentBlock) &&
+              parentBlock.getNextBlock() === draggable[0]) {
+            this.connectionDBList.push([parentBlock.nextConnection,
+              draggable[0].previousConnection]);
+          }
+        }
+      } else {
+        // If the draggable is not a block (e.g. ws comment)
+        this.topSubDraggables.push(draggable[0]);
+      }
+
+      this.subDraggables.set(draggable[0],
+          draggable[0].getRelativeToSurfaceXY());
+    }
+    Blockly.Events.setGroup();
```

Comment:
This will not work, put `Blockly.Events.setGroup(true);` at the beginning of `startDrag(e)`, and `Blockly.Events.setGroup(false);` at the end of `endDrag(e)`, also check if we are in a group or not (Just directly copy from https://github.com/google/blockly/pull/8355/files#diff-a0d82110fb1a9c53ea94842f102f20e07cd6ce1c721a88cd64e6b5911087522d)